src/main/java/org/folio/circulation/domain/anonymization/service/AnonymizationCheckersProvider.java

```diff
@@ -0,0 +1,101 @@
+package org.folio.circulation.domain.anonymization.service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.folio.circulation.domain.anonymization.checkers.AnonymizationChecker;
+import org.folio.circulation.domain.anonymization.checkers.AnonymizeLoansImmediatelyChecker;
+import org.folio.circulation.domain.anonymization.checkers.AnonymizeLoansWithFeeFinesImmediatelyChecker;
+import org.folio.circulation.domain.anonymization.checkers.FeesAndFinesClosePeriodChecker;
+import org.folio.circulation.domain.anonymization.checkers.LoanClosePeriodChecker;
+import org.folio.circulation.domain.anonymization.checkers.NeverAnonymizeLoansChecker;
+import org.folio.circulation.domain.anonymization.checkers.NeverAnonymizeLoansWithFeeFinesChecker;
+import org.folio.circulation.domain.anonymization.checkers.NoAssociatedFeesAndFinesChecker;
+import org.folio.circulation.domain.anonymization.config.LoanHistorySettings;
+
+import com.google.inject.internal.util.Lists;
+
+class AnonymizationCheckersProvider {
+
+  private final LoanHistorySettings loanHistorySettings;
+  private final List<AnonymizationChecker> generalCheckers;
+  private final List<AnonymizationChecker> feesAndFinesCheckers;
+  private final List<AnonymizationChecker> closedLoansCheckers;
+
+  AnonymizationCheckersProvider(LoanHistorySettings settings) {
+    this.loanHistorySettings = settings;
+    generalCheckers = generalCheckers();
+    feesAndFinesCheckers = feesAndFinesCheckers();
+    closedLoansCheckers = closedLoansCheckers();
+  }
+
+  AnonymizationCheckersProvider() {
+    this(null);
+  }
+
+  private List<AnonymizationChecker> generalCheckers() {
+    return Collections.singletonList(new NoAssociatedFeesAndFinesChecker());
+  }
+
+  private List<AnonymizationChecker> closedLoansCheckers() {
+    List<AnonymizationChecker> result = new ArrayList<>();
+    if (loanHistorySettings == null) {
+      return result;
+    }
+
+    switch (loanHistorySettings.getLoanClosingType()) {
+    case IMMEDIATELY:
+      result.add(new AnonymizeLoansImmediatelyChecker());
+      break;
+    case INTERVAL:
+      result.add(new LoanClosePeriodChecker(loanHistorySettings.getLoanClosePeriod()));
+      break;
+    case UNKNOWN:
+    case NEVER:
+      result.add(new NeverAnonymizeLoansChecker());
+      break;
+    default:
+      return result;
+    }
```

Comment:
Is this speculative design then? Currently, only a single checker applies in any given circumstance?

Reply:
Currently, it looks like that.

Reply:
One checker for closed loans without fees and fines, one checker for closed loans WITH fees and fines, one for manual anonymization.
It was initially a single list with all those checkers; hovewer, when the work started on the "never settings" - the new requirements, the list was split into three.

Reply:
I think I'm a little lost. Does this list contain all of the possible checkers, or only ever a single checker appropriate to the situation?

Reply:
Ok right now, it's three lists with a single checker appropriate to the situation. The lists are technically excessive now, but simplify extensibility in the future,  should the requirements change.
Do you think the lists should be refactored?

Reply:
Personally, I think if there is only a single checker relevant for each situation, I would only provide a single checker rather than a list.

I understand the desire to create space in the design for future requirements. I personally think that is speculative, and adds potentially unnecessary complexity. 

If there is a need to combine multiple checkers in some scenarios in the future, this could be accomodated using something like the composite pattern without changing the client API (or at least not much, maybe there needs to be multiple reasons).

Reply:
The code has been refactored.